### DISCUSSION TITLE
sql adding a new row to select query

### DISCUSSION THREAD

**Discussion by Manaralg** [sqlpage/SQLPage#582] [Q&A]
how to add an average new row to a select query?
 I have a select query that has 50 rows and I want to add a row on top of all rows named averages to calculate only 4 or the columns averages how do I do it?

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage! This is a good sql question. This can be solved using a [common table expression](https://en.m.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL) and an [UNION](https://en.m.wikipedia.org/wiki/Set_operations_(SQL)): 

```sql
WITH InitialQuery AS (
    -- Your complex query goes here
    SELECT column1, column2, column3
    FROM some_table
    WHERE some_condition
)

SELECT -- Return averages first
    AVG(column1) AS column1,
    AVG(column2) AS column2,
    AVG(column3) AS column3
FROM InitialQuery

UNION ALL -- Then return the full results

SELECT * FROM InitialQuery;
```
